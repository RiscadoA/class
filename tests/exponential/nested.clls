proc main() {
    cut {
        println("!a"); !a(x: close);
        println("close a"); close x
        |a: ?wait|
        println("?a"); ?a;
        cut {
            println("!b"); !b(x: close);
            println("call a"); call a(y);
            println("fwd"); fwd x y
            |b: ?wait|
            println("?b"); ?b;
            println("call 1"); call a(y);
            println("wait 1"); wait y;
            println("call 2"); call b(z);
            println("wait 2"); wait z;
            ()
        }
    }
};;
