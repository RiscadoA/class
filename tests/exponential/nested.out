!a
?a
!b
?b
call 1
close a
wait 1
call 2
call a
close a
fwd
wait 2
